pkg/golinters/ttempdir/ttempdir.go

```diff
@@ -0,0 +1,30 @@
+package ttempdir
+
+import (
+	"github.com/peczenyj/ttempdir/analyzer"
+	"golang.org/x/tools/go/analysis"
+
+	"github.com/golangci/golangci-lint/pkg/config"
+	"github.com/golangci/golangci-lint/pkg/goanalysis"
+)
+
+func New(settings *config.TtempdirSettings) *goanalysis.Linter {
+	a := analyzer.New()
+
+	var cfg map[string]map[string]any
+	if settings != nil {
+		cfg = map[string]map[string]any{
+			a.Name: {
+				analyzer.FlagAllName:               settings.All,
+				analyzer.FlagMaxRecursionLevelName: settings.MaxRecursionLevel,
+			},
+		}
+	}
+
+	return goanalysis.NewLinter(
+		a.Name,
+		"Detects the use of os.MkdirTemp, ioutil.TempDir or os.TempDir instead of t.TempDir",
```

Comment:
@peczenyj perhaps we could change [this line](https://github.com/peczenyj/ttempdir/blob/b4192abc1b673620de7d73eae0ed27c7a6770a6d/analyzer/analyzer.go#L16) to the `doc  = "Detects the use of os.MkdirTemp, ioutil.TempDir or os.TempDir instead of t.TempDir"` and use it here:

```suggestion
		a.Doc,
```

Reply:
I added this line: I replaced the `a.doc` by a better description.